src/Nancy/HttpLink.cs

```diff
@@ -180,7 +185,22 @@ public bool Equals(HttpLink other)
         /// </returns>
         public override bool Equals(object other)
         {
-            return ((IEquatable<HttpLink>)this).Equals(other as HttpLink);
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))

```

Comment:
Both this and the null check is covered in the other equals method. I think `return other is HttpLink && this.Equals((HttpLink) other)` should be enough here...